homeassistant/components/screenlogic/binary_sensor.py

```diff
@@ -104,67 +220,58 @@ async def async_setup_entry(
         config_entry.entry_id
     ]
     gateway = coordinator.gateway
-    data_path: ScreenLogicDataPath
-    value_params: SupportedBinarySensorValueParameters
-    for data_path, value_params in iterate_expand_group_wildcard(
-        gateway, SUPPORTED_DATA
-    ):
-        entity_key = generate_unique_id(*data_path)
-
-        device = data_path[0]
-
-        if not (DEVICE_INCLUSION_RULES.get(device) or value_params.included).test(
-            gateway, data_path
-        ):
-            cleanup_excluded_entity(coordinator, DOMAIN, entity_key)
-            continue
 
-        try:
-            value_data = gateway.get_data(*data_path, strict=True)
-        except KeyError:
-            _LOGGER.debug("Failed to find %s", data_path)
+    entities.extend(
+        [
+            ScreenLogicPushBinarySensor(coordinator, core_sensor_description)
+            for core_sensor_description in SUPPORTED_CORE_SENSORS
+        ]
+    )
+
+    pump_sensor_description: ScreenLogicBinarySensorDescription
+    for pnum, pdata in gateway.get_data(DEVICE.PUMP).items():
+        if pdata[VALUE.DATA] == 0:
             continue
+        pump_type = pdata[VALUE.TYPE]
+        for proto_pump_sensor_description in SUPPORTED_PUMP_SENSORS:
+            pump_sensor_description = copy(proto_pump_sensor_description)
+            device, group, value = pump_sensor_description.data_path
+            group = int(pnum)
+            pump_sensor_description.data_path = (device, group, value)
```

Comment:
Move out the parts that are not specific for the description, ie group, unique_id generation and name. Those don't have to be set on the entity description but can be set in the entity.

Reply:
Do you mean apply them as kwargs in the class constructor, or pull them all the way into [ScreenLogicEntity's init](https://github.com/home-assistant/core/blob/75951dd67be7c2e149348c6171a9af5b07d4de8a/homeassistant/components/screenlogic/entity.py#L45-L64)? (which could be possible)

Reply:
If all the entities share the parameters we can accept them and set the features in the base class init method. Otherwise do it in each platform entity init method.

Reply:
[77394cc](https://github.com/home-assistant/core/pull/100173/commits/77394cc35ab100aba66d6d8ef60c5daea99d6ef0)